dexs/raindex/index.ts

```diff
@@ -0,0 +1,148 @@
+import { CHAIN } from "../../helpers/chains"
+import { Interface, id, EventLog } from "ethers"
+import { BaseAdapter, BreakdownAdapter, FetchV2 } from "../../adapters/types"
+
+const orderbooks: Record<string, { address: string, start: number }> = {
+  [CHAIN.ARBITRUM]: {
+    address: "0x550878091b2b1506069f61ae59e3a5484bca9166",
+    start: 1727110056,
+  },
+  [CHAIN.BASE]: {
+    address: "0xd2938e7c9fe3597f78832ce780feb61945c377d7",
+    start: 1724856007,
+  },
+  [CHAIN.BSC]: {
+    address: "0xd2938e7c9fe3597f78832ce780feb61945c377d7",
+    start: 1727110200,
+  },
+  [CHAIN.FLARE]: {
+    address: "0xcee8cd002f151a536394e564b84076c41bbbcd4d",
+    start: 1725430973,
+  },
+  [CHAIN.LINEA]: {
+    address: "0x22410e2a46261a1b1e3899a072f303022801c764",
+    start: 1727718941,
+  },
+  [CHAIN.POLYGON]: {
+    address: "0x7d2f700b1f6fd75734824ea4578960747bdf269a",
+    start: 1726792922,
+  },
+} as const
+
+const ABI = {
+  AfterClear: "event AfterClear(address sender, (uint256 aliceOutput, uint256 bobOutput, uint256 aliceInput, uint256 bobInput) clearStateChange)",
+  TakeOrderV2: "event TakeOrderV2(address sender, ((address owner, (address interpreter, address store, bytes bytecode) evaluable, (address token, uint8 decimals, uint256 vaultId)[] validInputs, (address token, uint8 decimals, uint256 vaultId)[] validOutputs, bytes32 nonce) order, uint256 inputIOIndex, uint256 outputIOIndex, (address signer, uint256[] context, bytes signature)[] signedContext) config, uint256 input, uint256 output)",
+  ClearV2: "event ClearV2(address sender, (address owner, (address interpreter, address store, bytes bytecode) evaluable, (address token, uint8 decimals, uint256 vaultId)[] validInputs, (address token, uint8 decimals, uint256 vaultId)[] validOutputs, bytes32 nonce) alice, (address owner, (address interpreter, address store, bytes bytecode) evaluable, (address token, uint8 decimals, uint256 vaultId)[] validInputs, (address token, uint8 decimals, uint256 vaultId)[] validOutputs, bytes32 nonce) bob, (uint256 aliceInputIOIndex, uint256 aliceOutputIOIndex, uint256 bobInputIOIndex, uint256 bobOutputIOIndex, uint256 aliceBountyVaultId, uint256 bobBountyVaultId) clearConfig)",
+} as const
+
+const abi = new Interface([ ABI.ClearV2, ABI.AfterClear, ABI.TakeOrderV2 ])
+
+const fetchIntVol: FetchV2 = async function({ createBalances, api, fromTimestamp, toTimestamp }) {
```

Comment:
there's no reason to distinguish between "internal" and "external" volume, please remove that